app/models/event_invoice.py

```diff
@@ -72,3 +59,7 @@ def get_invoice_number(self):
 
     def __repr__(self):
         return '<EventInvoice %r>' % self.invoice_pdf_url
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.identifier = get_new_identifier(EventInvoice)
```

Comment:
This shouldn't be needed

Reply:
only making this call: get_new_identifier(EventInvoice) work?

Reply:
No, you need to create a new function which takes a model as an argument and returns a function (callable) which takes no argument, like previous get_new_identifier, and use that function in the default instead

Reply:
https://www.python-course.eu/currying_in_python.php

Reply:
what do you mean by last line?

> use that function in the default instead

Reply:
default=get_new_identifier(EventInvoice)